### DISCUSSION TITLE
Simple example of how to update checkbox in ag-Grid?

### DISCUSSION THREAD

**Discussion by garyrob** [elimintz/justpy#446] [Q&A]
UPDATE: I am now trying another approach, not using JustPy. So feel free to simply delete this issue. However, if you do have an answer that works well, it might motivate me to come back to JustPy, depending on how far I've gotten into the other approach.
--------------------------------------

The example at https://justpy.io/grids_tutorial/checkbox/ shows how to create a table with checkboxes and to perform an action when the state of a checkbox changes.

In my application, the user checks a checkbox (setting it to enabled=True) to indicate an action to be manually performed later by the user; when it's been performed, the checkbox should switch back to the enabled=False state.

I would like to be able to accomplish that in my code without the user having to reload the page in his browser. But I've spent a few hours trying to change the state of a checkbox in an ag-Grid from code after the page has been rendered, to no avail. 

A completely minimal example would be to change the grid_change() in the example code referred to above so that after the script prints `msg` to the console when the user clicks on a checkbox, that checkbox is changed back to its original state in the user's browser. That would not be a great UI design(!), but at least it would show how to change the checkbox's state.

Is there a way to do that? Or is that not something ag-Grid allows?

One thing I noticed in working on this is that run_api() doesn't seem to do anything for certain functions such as refreshCells() and redrawRows().

**Comment by WolfgangFahl** [comment]
@garyrob what did you choose as a work-around?

**Reply by garyrob** [reply]
Sorry, I ended up just doing other things and never got back to this. I'm not in a situation where I can go back to it now. I'm  basically just using Django and planning to use Unicorn or HTMX in places where I want more client-side interactivity than Django provides out of the box.
